snmp/tests/test_e2e_agent_ad.py

```diff
@@ -0,0 +1,147 @@
+# (C) Datadog, Inc. 2019-present
+# All rights reserved
+# Licensed under Simplified BSD License (see LICENSE)
+
+import pytest
+from tests.metrics import (
+    IF_BANDWIDTH_USAGE,
+    IF_COUNTS,
+    IF_GAUGES,
+    IF_RATES,
+    IP_COUNTS,
+    IP_IF_COUNTS,
+    IPX_COUNTS,
+    TCP_COUNTS,
+    TCP_GAUGES,
+    UDP_COUNTS,
+)
+
+from . import common
+
+pytestmark = pytest.mark.e2e
+
+
+def _build_device_ip(container_ip, last_digit='1'):
+    last_digit = str(last_digit)
+    snmp_device = container_ip.split('.')
+    snmp_device[len(snmp_device) - 1] = last_digit
+    snmp_device = '.'.join(snmp_device)
+    return snmp_device
+
+
+@common.agent_autodiscovery_only
+def test_e2e_agent_autodiscovery(dd_agent_check, container_ip, autodiscovery_ready):
+    """
+    Test Agent Autodiscovery
+
+    The assertions match `snmp_listener` configuration in `datadog.yaml`.
+    See `dd_environment` setup in `conftest.py`.
+    """
+    snmp_device = _build_device_ip(container_ip)
+    subnet_prefix = ".".join(container_ip.split('.')[:3])
+    aggregator = dd_agent_check({'init_config': {}, 'instances': []}, rate=True)
+
+    # === network profile ===
+    common_tags = [
+        'snmp_profile:generic-router',
+        'snmp_device:{}'.format(snmp_device),
+        'autodiscovery_subnet:{}.0/29'.format(subnet_prefix),
+        'tag1:val1',
+        'tag2:val2',
```

Comment:
Moving the whole test to a separate file for better separation. The only change is

```
        'tag1:val1',
        'tag2:val2',
```

Added to `common_tags`.